librato/instruments.py

```diff
@@ -55,14 +55,53 @@ def save(self):
 
 
 class Stream(object):
-    def __init__(self, metric=None, source='*', composite=None):
+    def __init__(self, metric=None, source='*', composite=None, name=None,
+            units_short=None, units_long=None, display_min=None, display_max=None,
+            summary_function='average', transform_function=None, period=None,
+            group_function='average', color=None):
```

Comment:
You can explore the option of iterating over the named arguments and set them in the object based on a class property that contains the valid list of stream attributes.